### PR TITLE
docs(add observables to TOH http)

### PR DESCRIPTION
**Pull request by thelgevold**
@wardbell @johnpapa 
This PR contains my addition of Observables to the TOH Http chapter

### CODE REVIEW

**Comment by thelgevold** [comment]
@wardbell 
Removing the WIP label. I have rebased with master and tested with the new version of in-memory-web-api. 
The chapter is ready for review.

**Comment by thelgevold** [comment]
I resolved merge conflicts from other PRs. This one is ready to go

File: public/docs/ts/latest/tutorial/toh-pt6.jade [hunk 1]
```diff
@@ -110,8 +110,8 @@ code-example(format="." language="bash").
 
   The Angular `http.get` returns an RxJS `Observable`.
   *Observables* are a powerful way to manage asynchronous data flows.
-  We'll learn about `Observables` *later*.
-
+  We'll learn about `Observables` in the next chapter.
```
Comment (chalin) [inline public/docs/ts/latest/tutorial/toh-pt6.jade]: In this line, and the block of added prose below change "chapter" to "section".

File: public/docs/ts/latest/tutorial/toh-pt6.jade [hunk 2]
```diff
@@ -305,6 +305,104 @@ figure.image-display
...
+ 
+ In this section we discuss `Observables` as an alternative to promises when processing http calls.
+
+ `Http` calls return RxJS observables by default, but so far we've been hiding that by converting the observable to a promise by calling `toPromise`.
+ 
+ Observables and promises are both great for processing http calls, but as we will see, the observables api is much richer. 
+ 
+ `RxJs` offers a wide variety of operators that we can use to manage event flows.     
+ 
+ In this chapter we will discuss how to use some of these operators to build a type-to-search filter where we can search for heroes by name.
```
Comment (chalin) [inline public/docs/ts/latest/tutorial/toh-pt6.jade]: Change "chapter" to "section".

**Comment by thelgevold** [comment]
@chalin Good call. I agree

**Comment by thelgevold** [comment]
Changed from chapter to section based on feedback from @chalin

**Comment by thelgevold** [comment]
Rebased to latest master

**Comment by thelgevold** [comment]
Rebased to latest master

**Comment by chalin** [comment]
@wardbell, @Foxandxss : I think that it would be nice if this were merged before the next wave(s) of changes hit the ToH. It is a well contained addition that would nicely round up the http toh chapter. (Btw, we should probably mark parts of the server-communication chapter as outdated, or just trim them out, because this ToH chapter will have outpaced it!)

**Comment by Foxandxss** [comment]
Will review soon!

**Comment by wardbell** [comment]
I made a bunch of clarifying edits in a separate commit.

Ready to squash and merge

**Pull request closed without merging**